src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/JsonFormat/SearchResultPackagesConverter.cs

```diff
@@ -31,13 +33,16 @@ public override void Write(Utf8JsonWriter writer, IPackageSearchMetadata value,
             writer.WriteStartObject();
             writer.WriteString(JsonProperties.PackageId, value.Identity.Id);
 
-            if (_exactMatch)
-            {
-                writer.WriteString(JsonProperties.Version, value.Identity.Version.ToNormalizedString());
-            }
-            else
+            if (value.Identity.Version is not null)
             {
-                writer.WriteString(JsonProperties.LatestVersion, value.Identity.Version.ToNormalizedString());
+                if (_exactMatch)
+                {
+                    writer.WriteString(JsonProperties.Version, value.Identity.Version.ToNormalizedString());
```

Comment:
It might be worth having an extension method:
```c#
public static void WriteStringIsNotNullOrWhiteSpace(this Utf8JsonWriter writer, string name, string? value)
{
    if (!string.IsNullOrWhiteSpace(value))
    {
        writer.WriteString(name, value);
    }
}
```